astropy/cosmology/tests/test_core.py

```diff
@@ -426,6 +426,29 @@ def test_equivalent_nonflat(self, cosmo):
         assert cosmo.equivalent_nonflat.is_equivalent(cosmo)
         assert cosmo.is_equivalent(cosmo.equivalent_nonflat)
 
+    # ------------------------------------------------
+    # clone
+
+    def test_clone_to_nonflat_equivalent(self, cosmo):
+        """Test method ``.clone()``to_nonflat argument."""
+        # just converting the class
+        nc = cosmo.clone(to_nonflat=True)
+        assert isinstance(nc, cosmo._nonflat_cls_)
+        assert nc == cosmo.equivalent_nonflat
+
+    @abc.abstractmethod
+    def test_clone_to_nonflat_change_param(self, cosmo):
+        """
+        Test method ``.clone()`` changing a(many) Parameter(s).
+        Nothing here b/c no Parameters.
+        """
+        # send to non-flat
+        nc = cosmo.clone(to_nonflat=True)
+        assert isinstance(nc, cosmo._nonflat_cls_)
+        assert nc == cosmo.equivalent_nonflat
```

Comment:
Why does this test not do what its name suggests?

Reply:
Because ``FlatCosmologyMixinTest`` tests a class without any Parameter objects, so there's nothing to test except for ``to_nonflat=True``. This test is really done in ``FlatFLRWMixinTest. test_clone_to_nonflat_change_param``, which calls ``super`` to get to this test.  I've structured the tests similarly to the code, so each class is tested and contributes tests to its child classes.

Reply:
>  I've structured the tests similarly to the code, so each class is tested and contributes tests to its child classes.

That sounds reasonable, so it would be good to add that comment to the code.

Reply:
✅